### PR TITLE
Reactive Filter Values

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

This adds a new helper function for creating queries that respond to changes more intuitively for input components. Currently it has only been applied to `<Dropdown/>`

### Checklist

- ~~[ ] For UI or styling changes, I have added a screenshot or gif showing before & after~~
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by ItsMeBrianD** [comment]
Resolves #1458

**Comment by csjh** [comment]
I'm getting a flash of `query returned no results` error when I first select a new option from the Dropdown

**Comment by ItsMeBrianD** [comment]
> I'm getting a flash of query returned no results error when I first select a new option from the Dropdown

Where are you seeing this?

File: packages/core-components/src/lib/atoms/inputs/dropdown/Dropdown.svelte [hunk 1]
```diff
@@ -65,19 +65,16 @@
 		</span>
 	{:else}
 		<select
-			disabled={hasQuery && !$query.loaded}
-			on:change={(e) => ($inputs[name] = e.currentTarget.value)}
+			disabled={hasQuery && !$query?.loaded}
```
Comment (csjh) [inline packages/core-components/src/lib/atoms/inputs/dropdown/Dropdown.svelte:68]:
```suggestion
			disabled={hasQuery && !$query.loaded}
```
I don't think it should be possible for `$query` to be undefined if `hasQuery`?

**Review by csjh — DISMISSED** [review]
LGTM

**Comment by csjh** [comment]
Uh actually scratch that I just noticed the error dropdown isn't erroring?

**Comment by ItsMeBrianD** [comment]
> Uh actually scratch that I just noticed the error dropdown isn't erroring?

Looks like this was being caused by a non-unique input name; the Dropdown queries were getting conflated. Ideally we want some way of tracking input names to prevent 2 inputs from being registered with the same name (possibly id, that is prefixed by the input name). I think that is out of scope here though

**Review by csjh — APPROVED** [review]
a couple questions and nitpicks but otherwise LGTM

File: packages/component-utilities/src/buildQuery.js [hunk 2]
```diff
@@ -27,10 +28,40 @@ export const getQueryFunction = () => getContext(QUERY_CONTEXT_KEY);
...
+	return {
+		results: derived(internal, (v) => v),
+		update: async (queryProps) => {
+			const { hasQuery, query } = buildInputQuery(queryProps, id);
+			if (!hasQuery) {
+				internal.set({ hasQuery: false });
+			} else {
+				internal.update(async (currentQuery) => {
+					if (query.hash !== currentQuery) await query.fetch();
+					internal.set({ hasQuery, query });
```
[the remark this answers is not in the document]
Comment (ItsMeBrianD) [reply packages/component-utilities/src/buildQuery.js:53]: Copy paste strikes again

**Pull request merged**